src/librbd/io/Utils.h

```diff
@@ -60,9 +60,9 @@ void file_to_extents(ImageCtxT *image_ctx, uint64_t offset, uint64_t length,
                      striper::LightweightObjectExtents* object_extents);
 
 template <typename ImageCtxT = librbd::ImageCtx>
-void extent_to_file(ImageCtxT *image_ctx, uint64_t object_no, uint64_t offset,
-                    uint64_t length,
-                    std::vector<std::pair<uint64_t, uint64_t> >& extents);
+std::pair<Extents, ImageArea> extents_to_file(ImageCtxT* image_ctx,
+                                              uint64_t object_no,
+                                              const Extents& object_extents);
```

Comment:
> nit: `extents_to_file` name suggests that it is used to convert to data area.

Honestly, `extents_to_file` name doesn't suggest anything to me because it has always been misleading -- there are no files in RBD and the name was just inherited from its `Striper` counterpart.  The proper verbose name for this implementation would be `object_extents_to_image_area_extents`.  I didn't come up with a nice shorthand for that and just kept the old name since the concept remained the same -- but then I didn't think about it too much.  Do you have a suggestion?

> May be leave `extent_to_file` with the old signature and provide `extents_to_area` with this new signature, and `extent_to_file` would be just a wrapper around `extents_to_area`, and would still be used where we know it is data area.

That is something I definitely want to avoid because I didn't want to make the DATA area implicit.  Making all places that assume or "know" the area expilictly mention it so that one could grep for `ImageArea::` and get all of them was a goal.

Reply:
To me "file" means a higher abstraction layer for actual user data. So it was not problem to think about it as a synonym for "image". Anyway, if you don't like `extents_to_file` with the old signature, still using the name `extents_to_file` with the new signature looks confusing to me, because now I don't understand what "file" actually means here.

Reply:
Not sure I like these new names. And indeed, shorter name would be nicer. I think I still like `extents_to_area` more but I don't have a strong opinion here. I will be fine with just leaving `extent_to_file`. 

Another name that comes to my mind: `logical to physical`.

Reply:
`logical_to_physical` would mean something different: it would be a mapping from extent+area (an image extent with a base offset equal to the area offset) to a raw extent (an image extent with a base offset equal to 0).  A raw/physical extent can then be mapped to a set of object extents using `Striper`:
```
extent_to_objects(extent):
    raw_extent_to_objects(extent_to_raw_extent(extent))
```
The way I see it, `logical_to_physical` would be another name for `extent_to_raw_extent`, not for the entire operation.  IOW a physical extent is still in image extent, not a object extent.

Reply:
I went ahead and renamed:
- `file_to_extents` to `area_to_object_extents`
- `extents_to_file` to `object_to_area_extents`
- `remap_offset` overloads to `area_to_raw_offset` and `raw_to_area_offset` respetively